tests/test_functions.py

```diff
@@ -0,0 +1,119 @@
+import requests, difflib, sys
+import re
+
+command_line_arguments = []
+
+number_of_errors = 0
+
+
+def set_command_line_arguments(arguments):
+    global command_line_arguments
+    command_line_arguments = arguments
+    if "-resetalltests" in command_line_arguments:
+        test_print("resetting all tests by saving every request for future comparisons. You should not have run this unless you are working on the test suite implementation itself and have verified that all tests passed before the reset.")
+
+def test_print(string):
+    sys.stdout.write("[synbiohub test] ")
+    sys.stdout.write(string)
+    sys.stdout.write("\n")
+
+# make html a little more human readable
+def format_html(htmlstring):
+    # for two angle brackets, insert a new line
+    newstring = re.sub('><', '>\n<', htmlstring)
+    # for an ending angle bracket and then something else, enter a new line
+    newstring = re.sub('>(.)', r'>\n\1', newstring)
+    # for something else then an angle bracket, also insert a new line
+    newstring = re.sub('(.)<', r'\1\n<', newstring)
+    
+    return newstring
+
+# perform a get request
+def get_request(request):
+    response = requests.get('http://localhost:7777/' + request)
+    
+    response.raise_for_status()
+    
+    content = format_html(response.text)
+
+    return content
+
+# data is the data field for a request
+def post_request(request, data):
+    response = requests.post('http://localhost:7777/' + request, data = data)
+    response.raise_for_status()
+    
+    content = format_html(response.text)
+    return content
+
+# perform and save a get request in previous results
+def save_get_request(request):
+    content = get_request(request)
+
+    filepath = 'previousresults/getrequest_' + request + '.txt'
+    
+    with open(filepath, 'w') as rfile:
+        rfile.write(content)
+
+# creates a file path for a given request and request type
+def request_file_path(request, requesttype):
+    return 'previousresults/' + requesttype.replace(" ", "") + "_" + request + ".txt"
+        
+# check a request against previous results
+# request should the page that was requested, such as /setup
+# requesttype is the type of request performed, either "get request" or "post request"
+def compare_request(requestcontent, request, requesttype):
+    # if the global state is to replace all files, do that
+    if "-resetalltests" in command_line_arguments:
```

Comment:
a function called `compare_request` shouldn't also be writing the request to file

Reply:
For the command line option -resetalltests, all the files are saved instead of compared. Would you like me to rename the funtion?